new unit tests in dark-gravity-wave-test added

abvgedeika opened the pull request.

abvgedeika committed:
new unit dark-gravity-wave-test added

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid when num blocks < maxBlocks and maxTarget = 0x207fffff', () => {//TODO is it True?
```
yes as I understand if less than 25 blocks has been mined the maxtarget is valid.

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+        it('should be valid when num blocks < maxBlocks and maxTarget = 0x207fffff', () => {//TODO is it True?
+            const highTargetInt = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetInt, blocks.slice(0, 10));
+            expect(result).to.equal(true);// TODO change on false?
```
not sure what you mean by _change on false_?

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+        });
+
+        it('should throw error for allHeaders is string', () => {
+            // TODO not very good that we allow string although it's length < 25 and ignored in getTarget
```
yes agree.  maybe we should add some validation on the `allHeaders` param provided?

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+
+        it('height values ignored in calculation in getTarget function', () => {
+            //TODO is it true?
+            //and we can pass any blocks height: invalid, duplicates, not ordered...?
```
yes I think that is fine.  Consensus rules do not consider height and its up to the dev to provide correct `allHeaders`.  Since DGW is stateless it cannot determine if the previous 25 headers provided is conforming to consensus based on earlier blocks.

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+            expect(result).to.equal(false);
+        });
+
+        it('should be invalid for blockTime negative', () => {//TODO We should not allow negative blockTime
```
agree

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+        });
+
+        it('should be invalid when timestamp for some blocks grater then next one', () => {//TODO We should check if timestamp is grater then previous
+            const highTargetBits = 0x1b0777d4;
```
more or less similar reasoning as in line 209.  however we might consider notifying the dev with an exception if things  like this is picked up.  Then again for `timestamp` in specific I am guessing that there is edge cases where timestamps can be out of order and still valid - Im not really sure.  Leaning towards not checking these fields.

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+            expect(result).to.equal(true);//TODO change on false!
+        });
+
+        it('should be invalid when timestamp for some blocks grater then next one', () => {//TODO We should check if timestamp is grater then previous?
```
similar as per prev comment ^

pvrooyen commented on test/dark-gravity-wave-test.js:
```diff
@@ -1,126 +1,286 @@
+        });
+
+
+        it('should be invalid when reverse block array', () => {//TODO Should we check order of blocks?
```
similar as per line 209

pvrooyen reviewed (approved):
utAck with comments

pvrooyen reviewed (dismissed):
utAck with comments

abvgedeika committed:
add new tests in dark-gravity-wave-test.js

+ link 2 issues:
https://dashpay.atlassian.net/browse/EV-897
https://dashpay.atlassian.net/browse/EV-898

abvgedeika committed:
run with travis

abvgedeika committed:
fix based on eslint formats for travis

abvgedeika committed:
minor chnages in test/dark-gravity-wave-test.js

pvrooyen reviewed (approved):
utAck
